File: Python/PE-3.py
# Largest prime factor
#
# https://projecteuler.net/problem=3

def primeFactors(n):
    i = 2
    factors = []
    while i * i <= n:
        if n % i:
            i += 1
        else:
            n //= i
            factors.append(i)
    if n > 1:
        factors.append(n)
    return factors

print(primeFactors(600851475143))
print(max(primeFactors(600851475143)))

File: C/PE_3.c
// Largest prime factor
//
// https://projecteuler.net/problem=3

#include <stdio.h>

int main() {
    long int n = 600851475143;

    int i = 2;
    int largest = 0;
    while (i * i <= n) {
        if (n % i != 0) {
            i++;
        }
        else {
            n /= i;
            largest = i;
        }
    }
    if (n > largest) largest = n;

    printf("%d\n", largest);

    return 0;
}
